providers/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java

```diff
@@ -351,14 +351,8 @@ public void suspendNode(final String id) {
    @Override
    public Iterable<VirtualMachine> listNodes() {
       ImmutableList.Builder<VirtualMachine> nodes = builder();
-      for (ResourceGroup resourceGroup : api.getResourceGroupApi().list()) {
-         List<VirtualMachine> vms = api.getVirtualMachineApi(resourceGroup.name()).list();
-         nodes.addAll(filter(vms, new Predicate<VirtualMachine>() {
-            @Override
-            public boolean apply(VirtualMachine input) {
-               return regionIds.get().isEmpty() || regionIds.get().contains(input.location());
-            }
-         }));
+      for (final String location : regionIds.get()) {
```

Comment:
If `regionIds` is empty (no region constraint) this should return all nodes. It currently returns nothing.